src/battle/overlay_12_0224E4FC.c

```diff
@@ -1770,3 +1770,195 @@ void ov12_02251038(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+        ctx->moveNoCopiedHit[i][battlerId] = 0;
+    }
+    
+    ov12_02258584(ctx, battlerId);
+    ov12_0225859C(ctx, battlerId);
+    ov12_022585A8(ctx, battlerId);
+}
+
+void InitFaintedWork(BattleSystem *bsys, BATTLECONTEXT *ctx, int battlerId) {
+    int i;
+    int maxBattlers;
+    u8 *data;
+    
+    maxBattlers = BattleSys_GetMaxBattlers(bsys);
+
+    for (int stat = 0; stat < 8; stat++) {
+        ctx->battleMons[battlerId].statChanges[stat] = 6;
+    }
+    
+    ctx->battleMons[battlerId].status2 = 0;
+    ctx->battleMons[battlerId].moveEffectFlags = 0;
+    
+    for (i = 0; i < maxBattlers; i++) {
+        if ((ctx->battleMons[i].status2 & STATUS2_MEAN_LOOK) && ctx->battleMons[i].unk88.battlerIdMeanLook == battlerId) {
+            ctx->battleMons[i].status2 &= ~STATUS2_MEAN_LOOK;
+        }
+        if (ctx->battleMons[i].status2 & (MaskOfFlagNo(battlerId) << STATUS2_ATTRACT_SHIFT)) {
+            ctx->battleMons[i].status2 &= (MaskOfFlagNo(battlerId) << STATUS2_ATTRACT_SHIFT) ^ 0xFFFFFFFF;
+        }
+        if ((ctx->battleMons[i].status2 & STATUS2_BINDING_ALL) && ctx->battleMons[i].unk88.battlerIdBinding == battlerId) {
+            ctx->battleMons[i].status2 &= STATUS2_BINDING_ALL ^ 0xFFFFFFFF;
+        }
+    }
+    
+    data = (u8 *)&ctx->battleMons[battlerId].unk88;
+    for (i = 0; i < sizeof(UnkBattlemonSub); i++) {
+        data[i] = 0;
+    }
+    
+    data = (u8 *)&ctx->turnData[battlerId];
+    for (i = 0; i < sizeof(TurnData); i++) {
+        data[i] = 0;
+    }
+    
+    ctx->battleMons[battlerId].unk88.fakeOutCount = ctx->totalTurns + 1;
+    ctx->battleMons[battlerId].unk88.slowStartTurns = ctx->totalTurns + 1;
+    ctx->battleMons[battlerId].unk88.truantFlag = (ctx->totalTurns + 1) & 1;
+    
+    ctx->moveNoProtect[battlerId] = 0;
+    ctx->moveNoHit[battlerId] = 0;
+    ctx->moveNoHitBattler[battlerId] = 0xFF;
+    ctx->moveNoHitType[battlerId] = 0;
+    ctx->moveNoBattlerPrev[battlerId] = 0;
+    ctx->moveNoCopied[battlerId] = 0;
+    ctx->moveNoCopiedHit[battlerId][0] = 0;
+    ctx->moveNoCopiedHit[battlerId][1] = 0;
+    ctx->moveNoCopiedHit[battlerId][2] = 0;
+    ctx->moveNoCopiedHit[battlerId][3] = 0;
+    ctx->moveNoSketch[battlerId] = 0;
+    ctx->conversion2Move[battlerId] = 0;
+    ctx->conversion2BattlerId[battlerId] = 0;
+    ctx->conversion2Type[battlerId] = 0;
+    ctx->moveNoMetronome[battlerId] = 0;
+    
+    ctx->fieldCondition &= (MaskOfFlagNo(battlerId) << 8) ^ 0xFFFFFFFF; //??
+    
+    for (i = 0; i < maxBattlers; i++) {
+        if (i != battlerId && BattleSys_GetFieldSide(bsys, i) != BattleSys_GetFieldSide(bsys, battlerId)) {
+            ctx->moveNoCopied[i] = 0;
+        }
+        ctx->moveNoCopiedHit[i][battlerId] = 0;
+    }
+
+    ctx->unk_13C[battlerId] &= ~1;
+    
+    ov12_02258584(ctx, battlerId);
+    ov12_0225859C(ctx, battlerId);
+    ov12_022585A8(ctx, battlerId);
+}
+
+//BattleContext_InitTurnData..?
+void ov12_02251710(BattleSystem *bsys, BATTLECONTEXT *ctx) {
```

Comment:
rename?

Reply:
Still workshopping a name, since it does more than just init the TurnData struct